js/components/portal-dashboard/feedback/rubric-table.tsx

```diff
@@ -0,0 +1,155 @@
+    const ratingDescription =
+      (crit.ratingDescriptionsForStudent && crit.ratingDescriptionsForStudent[ratingId]) ||
+      (crit.ratingDescriptions && crit.ratingDescriptions[ratingId]) ||
+      null;
+    const isApplicableRating = crit.nonApplicableRatings === undefined ||
+      crit.nonApplicableRatings.indexOf(ratingId) < 0;
+    const isNotApplicable = !isApplicableRating;
+
+    return (
+      <div className={css.rubricScoreBox} key={radioButtonKey}>
+        <div className={css.rubricButton} title={(isApplicableRating) ? ratingDescription : "Not Applicable"}>
+          { isNotApplicable
+            ? <span className={css.noRating}>N/A</span>
+            : this.renderButton(critId, selected,ratingId, buttonIndex)
+          }
+        </div>
+      </div>
+    );
+  }
+
+  private renderButton = (critId: string, selected: boolean, ratingId: string, buttonIndex: number) => {
+    const handleRatingChange = (buttonIndex: number) => () => {
+      const { rubric, rubricFeedback } = this.props;
+      const deselect = rubric.ratings.findIndex((r: any) => r.id === (rubricFeedback[critId].id)) === buttonIndex;
+
+      updateSelection(critId, ratingId, deselect);
+    };
+
+    const updateSelection = (critId: any, ratingId: string, deselect: boolean) => {
+      const { rubric, rubricFeedback, student } = this.props;
+      const newSelection: any = {};
+      const rating = rubric.ratings.find((r: any) => r.id === ratingId);
+      const criteria = rubric.criteria.find((c: any) => c.id === critId);
+      const score = rating.score;
+      const label = rating.label;
+      const studentId = student.get("id");
+
+      deselect  ? newSelection[critId] = {
+                  id: "",
+                  score: 0,
+                  label: "",
+                  description: "",
+                }
+                : newSelection[critId] = {
+                  id: ratingId,
+                  score,
+                  label,
+                  description: criteria.ratingDescriptions[ratingId],
+                };
+      const newFeedback = Object.assign({}, rubricFeedback, newSelection);
+      this.rubricChange(newFeedback, studentId);
+    };
+
+    return (
+      <button className={css.outerCircle} onClick={handleRatingChange(buttonIndex)}>
+        <div className={`${css.innerCircle} ${selected ? css.selected : ""}`}></div>
+      </button>
+    );
+  }
+
+  private rubricChange = (rubricFeedback: any, studentId: string) => {
+    const { rubric, activityId, activityIndex, updateActivityFeedback } = this.props;
+    let numFeedback = 0;
+    rubric.criteria.map((crit: any)=>{
+      rubricFeedback[crit.id].id !== "" && numFeedback++;
+    });
+
+    const hasBeenReviewed  = numFeedback !== 0 ? true : false;
+    activityId && studentId && updateActivityFeedback && updateActivityFeedback(activityId, activityIndex, studentId, { rubricFeedback, hasBeenReviewed } );
```

Comment:
I would break this into multiple lines.  That said, if updateActivityFeedback isn't optional in the props list you can remove that check:
`    activityId && studentId && updateActivityFeedback(activityId, activityIndex, studentId, { rubricFeedback, hasBeenReviewed });`